File: subject/3주차/2.py
lst = list(map(int, input().split(' ')))
dis = int(input())
ret = []
def func(x, ans):
	if x >= dis:
		if x == dis:
			ret.append(ans)
		return 0
	for i in range(3):
		func(x + lst[i], ans + 1)
func(0,0)
if len(ret) == 0:
	print(-1)
else:
	print(min(ret))

File: 2020_2학기_알고리즘/3주차/2.cpp
#include<iostream>
using namespace std;

int step[4] = {0, };
int dp[4][10001] = {0, };
int dis = 0;

int func(int x, int ans){
	if (x >= dis)
		return 0;
	for (int i = 0; i < 3; i++){
		func(x + step[i], ans + 1);
	}
	return ans;
}

int main(){
	for (int i = 0; i < 3; i++){
		cin>>step[i];
	}
	cin>>dis;

	cout << func(0,0) << endl;
}
